### DISCUSSION TITLE
Add volume_flow_rate device class

### DISCUSSION THREAD

**Discussion by elupus** [home-assistant/architecture#915] [Entity Models] [answered]
## Proposal
Add sensor device class volume_flow_rate_device with supported units:

    - SI /metric
      - m³/h
      - L/min
    - USCS / imperial:
      - ft³/m
      - gal/m

## Background
* useful for keeping track instantaneous of water consumption in for example houses. Either by direct readout from integration which is supported on `knx`, `landis_gyr_heatmeter`,  `overkiz`, `zha`, `bthome` or as a result from derivation helper on volume device class sensors.
* required to support unit conversions between units, where scaling is required for useful presentation.

# Initial implementation
https://github.com/home-assistant/core/pull/106077

**Comment by TomBrien** [comment]
I like this. My only comment would be to use `min` instead of `m` for minute to avoid confusion should cubic metres per minute be added down the line. (This would also be in line with [SI unit system](https://www.npl.co.uk/si-units))

**Reply by elupus** [reply]
Indeed. Seems our current unit is wrong here: https://github.com/home-assistant/core/blob/efb92ca9eeb9d31bebc8493bfe8431c1aee0bd1f/homeassistant/const.py#L764

**Reply by elupus** [reply]
How should we handle the imperial units that are using the "wrong" "m" instead of "min" already?

**Reply by TomBrien** [reply]
I had a suspicion this would be the case. My opinion (and it is only that):

- Fix in code base. What's the pain here? Is it too much?
- Fix in frontend only (make it look right for the users and devs should be using constants which are clear enough)
- Carry this, all be it wrong, convention forward. The constant name is perfectly clear but it's a pity to have _possibly_ confuse users. 

If the first option isn't too painful in terms of breaking changes, I would favour that. Hopefully all core integrations use the constant and custom integrations are too difficult to consider. However, the second option seems a reasonable compromise.

**Reply by shbatm** [reply]
Support this as well, but wanted to comment that the standard abbreviations in USCS would usually be:

- Gallon per minute (gpm)
- Cubic foot per minute (cfm)
- Standard cubic feet per minute (scfm)
- Cubic foot per second (cfs)

**Reply by epenet** [reply]
> How should we handle the imperial units that are using the "wrong" "m" instead of "min" already?

Maybe EQUIVALENT_UNITS helps?
https://github.com/home-assistant/core/blob/85a12c37ee809720ff95854906592177900cd53e/homeassistant/components/sensor/recorder.py#L55

**Comment by jeeftor** [comment]
I think this would probably also benefit the Flume integration:

There have been issues open to let people change the units.
https://github.com/home-assistant/core/issues/82866#issuecomment-1670241103

To my quick look we are using "custom" units for the temporal based units:

<img width="584" alt="image" src="https://github.com/home-assistant/architecture/assets/6491743/7d52d3f0-db5e-4cc0-b8de-e4eb90062379">


And I'm assuming you cant change these because they aren't built in to Home Assistant...

**Comment by petervk** [comment]
Would love to have this be supported by Home Assistant!

**Comment by emontnemery** [comment, marked as answer]
This proposal is fine, let's go ahead with the implementation.
The issues with "wrong" unit being used in some integrations can be sorted out in the implementation PR.

**Comment by frenck** [comment]
It has been merged in https://github.com/home-assistant/core/pull/106077

**Comment by parautenbach** [comment]
<s>I'd like to [update the docs](https://www.home-assistant.io/integrations/sensor/), since it looks like it was omitted as part of this change. I cannot find the actual device class label (text) though. Can someone point that out to me in order to submit a PR for the docs?</s>

Nevermind, I found it. Not sure how I missed it.
